wetterdienst/dwd/radar/api.py

```diff
@@ -0,0 +1,233 @@
+    - https://opendata.dwd.de/climate_environment/CDC/grids_germany/hourly/radolan/
+    - https://opendata.dwd.de/climate_environment/CDC/grids_germany/5_minutes/radolan/
+    """
+
+    def __init__(
+        self,
+        parameter: Union[str, RadarParameter],
+        site: Optional[RadarSite] = None,
+        format: Optional[RadarDataFormat] = None,
+        subset: Optional[RadarDataSubset] = None,
+        start_date: Optional[Union[str, datetime, RadarDate]] = None,
+        end_date: Optional[Union[str, datetime, timedelta]] = None,
+        time_resolution: Optional[Union[str, TimeResolution]] = None,
+        period_type: Optional[PeriodType] = None,
+    ) -> None:
+        """
+        :param parameter:       The radar moment to request
+        :param site:            Site/station if parameter is one of
+                                RADAR_PARAMETERS_SITES
+        :param format:          Data format (BINARY, BUFR, HDF5)
+        :param subset:          The subset (simple or polarimetric) for HDF5 data.
+        :param start_date:      Start date
+        :param end_date:        End date
+        :param time_resolution: Time resolution for RadarParameter.RADOLAN_CDC,
+                                either daily or hourly or 5 minutes.
+        :param period_type:     Period type for RadarParameter.RADOLAN_CDC
+        :return:                Nothing for now.
+        """
+
+        # Convert parameters to enum types.
+        self.parameter = RadarParameter(parameter)
+        self.site = parse_enumeration_from_template(site, RadarSite)
+        self.format = parse_enumeration_from_template(format, RadarDataFormat)
+        self.subset = parse_enumeration_from_template(subset, RadarDataSubset)
+        self.time_resolution = parse_enumeration_from_template(
+            time_resolution, TimeResolution
+        )
+        self.period_type = parse_enumeration_from_template(period_type, PeriodType)
+
+        # Sanity checks.
+        if self.parameter == RadarParameter.RADOLAN_CDC:
+
+            if time_resolution not in (
+                TimeResolution.DAILY,
+                TimeResolution.HOURLY,
+                TimeResolution.MINUTE_5,
+            ):
+                raise ValueError(
+                    "RADOLAN_CDC only supports daily, hourly and 5 minutes resolutions"
+                )
+
+        if start_date == RadarDate.LATEST:
+
+            # HDF5 folders do not have "-latest-" files.
+            if self.parameter == RadarParameter.RADOLAN_CDC:
+                raise ValueError("RADOLAN_CDC data has no '-latest-' files")
+
+            # HDF5 folders do not have "-latest-" files.
+            if self.format == RadarDataFormat.HDF5:
+                raise ValueError("HDF5 data has no '-latest-' files")
+
+        if start_date == RadarDate.CURRENT and not self.period_type:
+            self.period_type = PeriodType.RECENT
+
+        # Evaluate "RadarDate.MOST_RECENT" for "start_date".
```

Comment:
But can I ask for a single elevation at 15:03 and get the file for the 15:00 o clock scan?

Reply:
I would not have dared to ask this, but this would be the icing of the cake.

Reply:
I believe `RadarDate.CURRENT` is what you are asking for here.

Reply:
Not sure, if that's what @meteoDaniel wants, but would be nice to select only a specific sweep (of 00, 01, ..., 09) from the volume. As I said, I would not have dared to ask, since this could be added in a later PR (might get my hands dirty with that).

Reply:
Now I am getting it, you are aiming at a **single sweep** here. Hm, yeah. If this actually makes sense, I will see what I can do here.

Reply:
A bit more context why users might want this, here is one example for using only a single sweep from the volume:

With higher elevation scans (>15 deg) one can process so called quasi vertical profiles (QVP). This is just a median over the azimuth for all ranges for this particular sweep. Stacking several sweeps along time it can be displayed like a time vs. height plot.

Reply:
This has been implemented with 7334c338.